five-minute-serverless-backend/frontend.html

```diff
@@ -26,59 +26,32 @@
     </style>
   </head>
   <body>
-    <div>Type in some text and hit Enter:</div>
+    <div>Type in some text:</div>
     <input type="text" id="input"/>
     <pre id="output"></pre>
     <script>
-     const sendHttpRequest = async (method, url, data, headers={}) => {
-        return new Promise((resolve, reject) => {
-          const xhr = new XMLHttpRequest();
-          xhr.onload = () => {
-            const headers = {};
-            const strings = xhr.getAllResponseHeaders().split('\n');
-            for (const s of strings) {
-              if (s) {
-                const [key, val] = s.split(': ');
-                headers[key] = val;
-              }
-            }
-            resolve({
-              status: xhr.status,
-              headers,
-              body: xhr.responseText
-            });
-          };
-          xhr.onerror = () => reject('Network error');
-          xhr.open(method, url, true);
-          for (const key in headers) {
-            xhr.setRequestHeader(key, headers[key]);
-          }
-          xhr.send(data);
-        });
-      };
-
       const invoke = async (name, input) => {
         const url = `https://run.binaris.com/v2/run/${BINARIS_ACCOUNT_ID}/${name}`;
-        const data = JSON.stringify(input);
+        const body = JSON.stringify(input);
         const headers = { 'Content-Type': 'application/json' };
-        const res = await sendHttpRequest('post', url, data, headers);
-        const ct = res.headers['content-type'];
+        const res = await fetch(url, {
+          method: 'POST',
+          body,
+          headers,
+        });
+        const ct = res.headers.get('Content-Type');
         if (res.status !== 200 || !ct || !ct.startsWith('application/json')) {
           throw new Error(`Backend error: status=${res.status} response: ${res.body}`);
         }
-        return JSON.parse(res.body);
+        return await res.json();
       };
 
       const $input = document.getElementById('input');
       const $output = document.getElementById('output');
-      $input.addEventListener('keyup', async e => {
-        if (e.keyCode === 13 && $input.value) {
-          const value = await invoke('public_backend', $input.value);
-          $output.innerHTML += value + '\n';
-          $input.value = '';
-        }
+      $input.addEventListener('input', async e => {
+        $output.innerHTML = await invoke('public_backend', $input.value);
```

Comment:
Not requesting debounce, Rx, etc. since those should be familiar in front-end anyway.